File: 01-collatz/collatz.jl
function collatz(n)
    cnt = 0
    while n > 1
        n = (n % 2 != 0) ? 3n+1 : div(n, 2)
        cnt += 1
    end
    return cnt
end

n = parse(Int64, ENV["MAXN"])
total = 0
for i = 1:n
    global total += collatz(i)
end

println("sum=$total, avg=$(total/n)")

File: 01-collatz/collatz.pl
sub collatz {
    my $n = shift @_;
    my $cnt = 0;
    while ($n > 1) {
        $n = $n % 2 ? $n * 3 + 1 : $n / 2;
        $cnt++;
    }
    return $cnt;
}

$max = $ENV{'MAXN'};

$sum = 0;

for ($i = 1; $i <= $max; $i++) {
    $sum += collatz($i);
}

$avg = $sum / $max;

print "sum=$sum, avg=$avg\n";
